uniflow/op/model/model_server.py

```diff
@@ -866,6 +898,166 @@ def __call__(self, data: List[str]) -> List[str]:
         return data
 
 
+class SageMakerModelServer(AWSBaseModelServer):
+    """
+    SageMaker Model Server Class.
+
+    The AWS client authenticates by automatically loading credentials as per the methods outlined here:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If you wish to use a specific credential profile, please provide the profile name from your ~/.aws/credentials file.
+
+    Make sure that the credentials or roles in use have the necessary policies for SageMaker service access.
+    """
+
+    def __init__(
+        self, prompt_template: PromptTemplate, model_config: Dict[str, Any]
+    ) -> None:
+        super().__init__(prompt_template, model_config)
+        self._model_config = SageMakerModelConfig(**self._model_config)
+        self._client = self.session.client(
+            "sagemaker-runtime", region_name=self.aws_region
+        )
+
+    def prepare_input(
+        self, model_type: str, prompt: str, model_kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Prepare the input for the model based on the model_type.
+
+        Args:
+            model_type (str): The type of the model.
+            prompt (str): The input prompt.
+            model_kwargs (Dict[str, Any]): Additional model arguments.
+
+        Returns:
+            Dict[str, Any]: The prepared input for the model.
+        """
+
+        def prepare_falcon_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_body = {
+                "inputs": f"{prompt}",
+                "parameters": model_kwargs,
+            }
+            return input_body
+
+        def prepare_default_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_body = {"inputs": prompt, "parameters": model_kwargs}
+            return input_body
+
+        model_input_preparation = {
+            "falcon": prepare_falcon_input,
+            "mistral": prepare_default_input,
+        }
+
+        prepare_input_for_model = model_input_preparation.get(
+            model_type, prepare_default_input
+        )
+        return prepare_input_for_model(prompt, model_kwargs)
+
+    def prepare_output(self, model_type: str, response: Any) -> str:
+        """
+        Prepares the output based on the model_type and response.
+
+        Args:
+            model_type (str): The model_type of the response.
+            response (Any): The response object.
+
+        Returns:
+            str: The prepared output.
+
+        Raises:
+            None
+        """
+
+        def prepare_falcon_output(response: Any) -> str:
+            response_body = json.loads(response.get("Body").read())
+            return response_body[0].get("generated_text")
+
+        def prepare_default_output(response: Any) -> str:
+            response_body = json.loads(response.get("Body").read())
+            return response_body.get("outputs")
```

Comment:
nit: `default` can be a bit misleading. can we change to `mistral` to indicate this output parser is for mistral.

Reply:
Sure! Will do